tests/acceptance/features/loginUsingEmailAddress.feature

```diff
@@ -0,0 +1,76 @@
+@insulated @disablePreviews
+Feature: login users
+
+As an admin
+I want users to login with their email address stored on a LDAP server
+So that they only need to remember one username and password (SSO)
+
+	Scenario: login with default settings
+		When the LDAP users have been resynced
+		Then it should be possible to login with the username "user1@example.org" and password "1234" into the WebUI
+
+	Scenario: using ldap filter including email field 
+		When the LDAP config "LDAPTestId" has these settings:
+		|key                | value                                                                            |
+		|ldapLoginFilter    | (&(objectclass=inetOrgPerson)(\|(uid=%uid)(mailPrimaryAddress=%uid)(mail=%uid))) |
+		|ldapEmailAttribute |                                                                                  |
+		Then it should be possible to login with the username "user1@example.org" and password "1234" into the WebUI
+
+	Scenario: using ldapEmailAttribute but loginFilter lacks email field 
+		When the LDAP config "LDAPTestId" has these settings:
+		|key                | value                                    |
+		|ldapLoginFilter    | (&(objectclass=inetOrgPerson)(uid=%uid)) |
+		|ldapEmailAttribute | mail                                     |
+		Then it should be possible to login with the username "user1@example.org" and password "1234" into the WebUI
+
+	Scenario: no ldapEmailAttribute and loginFilter lacks email field 
+		When the LDAP config "LDAPTestId" has these settings:
+		|key                | value                                    |
+		|ldapLoginFilter    | (&(objectclass=inetOrgPerson)(uid=%uid)) |
+		|ldapEmailAttribute |                                          |
+		And the LDAP users have been resynced
+		Then it should not be possible to login with the username "user1@example.org" and password "1234" into the WebUI
+
+	Scenario: change Email address on LDAP server
+		When the admin sets the ldap attribute "mail" of the entry "uid=user1,ou=TestUsers" to "user1-change@example.org"
+		And the LDAP users have been resynced
```

Comment:
@individual-it it should actually be possible to login with the old and the new email, without syncing.

When the old emal is used oc will be able to find the user in the account table and then use the ldap backend to log him in using his username and the given password. the username will be mapped to the dn which will be used to check the password.

If the new email is used the user won't be found in the account table and oc will try to log in the user with every backend using the given credentials. The ldap login should work because the login filter will match the user with the new emal.

so both should work...

Reply:
added an extra test for changed email without sync https://github.com/owncloud/user_ldap/pull/183/files#diff-88d65b494ef1445d3203db3861bf3e56R42

Reply:
Awesome how quick you can react here - this is awesome progress!